packages/esm-patient-notes-app/src/notes/notes-detailed-summary.component.tsx

```diff
@@ -1,148 +1,171 @@
+      if (visit && visit.status === 'ongoing') {
+        setActiveVisit(visit);
+      }
+    });
+    return () => sub.unsubscribe();
+  }, []);
 
-      return () => subscription.unsubscribe();
+  React.useEffect(() => {
+    if (patientUuid) {
+      const sub = getEncounterObservableRESTAPI(patientUuid).subscribe(setNotes, (err) => {
+        setError(err);
+        createErrorHandler();
+      });
+
+      return () => sub.unsubscribe();
     }
-  }, [patientUuid, patient]);
+  }, [patientUuid]);
 
-  return (
-    <>
-      {patientNotes && (
-        <div className={styles.notesSummary}>
-          {patientNotes.length > 0 ? (
-            <SummaryCard
-              name={t('notes', 'Notes')}
-              addComponent={showAddNote}
-              showComponent={() => attach('patient-chart-workspace-slot', 'visit-notes-workspace')}>
-              <table className={`omrs-type-body-regular ${styles.notesTable}`}>
-                <thead>
-                  <tr className={styles.notesTableRow} style={{ textTransform: 'uppercase' }}>
-                    <th>{t('date', 'Date')}</th>
-                    <th style={{ textAlign: 'left' }}>
-                      {t('note', 'Note')}
-                      <svg
-                        className="omrs-icon"
-                        style={{
-                          height: '0.813rem',
-                          fill: 'var(--omrs-color-ink-medium-contrast)',
-                        }}>
-                        <use xlinkHref="#omrs-icon-arrow-downward"></use>
-                      </svg>
-                      <span style={{ marginLeft: '1.25rem' }}>{t('location', 'Location')}</span>
-                    </th>
-                    <th style={{ textAlign: 'left' }}>{t('author', 'Author')}</th>
-                    <th></th>
-                  </tr>
-                </thead>
-                <tbody>
-                  {pagination.results.map((note) => {
-                    return (
-                      <Fragment key={note.id}>
-                        <tr className={styles.notesTableDataRow}>
-                          <td className={styles.noteDate}>{formatDate(note?.encounterDate)}</td>
-                          <td className={styles.noteInfo}>
-                            <span className="omrs-medium">{note.encounterType}</span>
-                            <div
-                              style={{
-                                color: 'var(--omrs-color-ink-medium-contrast)',
-                                margin: '0rem',
-                              }}>
-                              {capitalize(note.encounterLocation)}
-                            </div>
-                          </td>
-                          <td className={styles.noteAuthor}>
-                            {note.encounterAuthor ? note.encounterAuthor : '\u2014'}
-                          </td>
-                          <td
+  const toggleShowAllNotes = () => {
```

Comment:
Can be made a `useCallback` without deps (if my recommendation below is followed, i.e., not capture the state)